src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -12,12 +12,9 @@
 
 @Dao
 public class MovieDaoImpl implements MovieDao {
-    @Override
     public Movie add(Movie movie) {
         Transaction transaction = null;
-        Session session = null;
-        try {
-            session = HibernateUtil.getSessionFactory().openSession();
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
The `Session` is correctly managed using a try-with-resources statement, which ensures it is closed automatically. However, ensure that the `Transaction` is also managed properly within this block.